src/main/java/com/hubspot/jinjava/lib/filter/NumberFormatFilter.java

```diff
@@ -0,0 +1,110 @@
+package com.hubspot.jinjava.lib.filter;
+
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableMap;
+import com.hubspot.jinjava.doc.annotations.JinjavaDoc;
+import com.hubspot.jinjava.doc.annotations.JinjavaParam;
+import com.hubspot.jinjava.doc.annotations.JinjavaSnippet;
+import com.hubspot.jinjava.interpret.JinjavaInterpreter;
+import com.hubspot.jinjava.interpret.TemplateError;
+import com.hubspot.jinjava.interpret.TemplateError.ErrorItem;
+import com.hubspot.jinjava.interpret.TemplateError.ErrorReason;
+import com.hubspot.jinjava.interpret.TemplateError.ErrorType;
+import java.math.BigDecimal;
+import java.text.DecimalFormat;
+import java.text.NumberFormat;
+import java.util.Locale;
+import java.util.Objects;
+
+@JinjavaDoc(
+  value = "Formats a given number based on the locale passed in as a parameter.",
+  input = @JinjavaParam(
+    value = "value",
+    desc = "The number to be formatted based on locale",
+    required = true
+  ),
+  params = {
+    @JinjavaParam(
+      value = "locale",
+      desc = "Locale in which to format the number. The default is the page's locale."
+    ),
+    @JinjavaParam(
+      value = "decimal precision number",
+      type = "number",
+      desc = "A number input that determines the decimal precision of the formatted value. If the number of decimal digits from the input value is less than the decimal precision number, use the number of decimal digits from the input value. Otherwise, use the decimal precision number. The default is the number of decimal digits from the input value."
+    )
```

Comment:
This is really a maximum, right? I think naming it as such would be clearer, something like `max decimal precision`.